ide-support/revsaveasiosstandalone.livecodescript

```diff
@@ -1507,13 +1527,35 @@ private command revCreateMobilePlist pSettings, pAppBundle, pTarget, pFonts, pPl
    end if
    replace "${LOCATION_AUTH_TYPE}" with it in pPlist
    
+   -- AB : helper stmts to generate the actual strings that we are going to insert 
+   -- the background modes into the plist file
+   
+   local tBackgroundValues
+   put "<string>audio</string>" into tBackgroundValues["${PLAY_AUDIO_WHEN_IN_BACKGROUND}"]
+   put "<string>location</string>" into tBackgroundValues["${BACKGROUND_LOCATION_UPDATE}"]
+   put "<string>voip</string>" into tBackgroundValues["${BACKGROUND_VOIP}"]
+   put "<string>newsstand-content</string>" into tBackgroundValues["${BACKGROUND_NEWSSTAND_DOWNLOADS}"]
+   put "<string>external-accessory</string>" into tBackgroundValues["${EXTERNAL_ACC_COMM}"]
+   put "<string>bluetooth-central</string>" into tBackgroundValues["${USE_BT_LE}"]
+   put "<string>bluetooth-peripherla</string>" into tBackgroundValues["${ACTS_AS_BT_LE}"]
+   put "<string>fetch</string>" into tBackgroundValues["${BACKGROUND_FETCH}"]
+   put "<string>remote-notification</string>" into tBackgroundValues["${REMOTE_NOTIFICATIONS}"]
+   
    -- PM-2015-10-08:  [[ Bug 16154 ]] Allow audio playing when the app is on background
-   if tBackgroundAudio is true then
-      get "<string>audio</string>"
-   else -- false or empty
-      get empty
-   end if
-   replace "${PLAY_AUDIO_WHEN_IN_BACKGROUND}" with it in pPlist
+   -- AB-2017-05-31: Add BackgroundModes to Plist
+   repeat for each key tKey in tBackgroundModes
+      if tBackgroundModes[tKey] is true then
+         replace tKey with tBackgroundValues[tKey] in pPlist
+      end if
```

Comment:
If for a `tKey` the `tBackgroundModes[tKey]` is not true, you have to remove the `tKey` from the plist:

```
if tBackgroundModes[tKey] is true then
      replace tKey with tBackgroundValues[tKey] in pPlist
else
     replace tKey with empty in pPList
end if
```